torchaudio/_internal/module_utils.py

```diff
@@ -76,6 +76,30 @@ def wrapped(*args, **kwargs):
             return wrapped
     return decorator
 
+def is_soundfile_available():
+    if not is_module_available('soundfile'): return False
+    try:
+        import soundfile
+    except OSError as os_error:
+        if str(os_error).find("sndfile library not found") != -1:
```

Comment:
Note os_error.error_no is None thus I was trying to use string match to check. pls let me know in case there is a better approach

Reply:
I think we can simply catch `Exception` and we do not need to check the detail of the exception.
The reason is that, from the perspective of `torchaudio` library, we only want to know if `soundfile` could be imported without error or not. If it was imported, then we enable some features in `torchaudio` and otherwise disable the features. We do not want to propagate the exception caught during `import soundfile` (we can add a warning if we want though), because for `torchaudio` it's optional dependency, and otherwise `import torchaudio` will stop working.

Because of the above reasons, it is not necessary for us to inspect the reason `import soundfile` failed. If `import soundfile` is necessary for client code, then we can expect that client code will run into a issue after `import torchaudio`, either they try to do `import soundfile` by themselves or try to use `torchaudio`'s functionality that requires `soundfile`. So it will not be that inconvenient for users even if we do not check what kind of exception was raised during `import soundfile`.

Reply:
Great point, modified the code to take OSError only, also updated error msg a little bit to make it precise.